### PR TITLE
Accomodate for the changes to SPI config introduced in PR 232

### PR DESCRIPTION
**Pull request by metlos**
This PR accomodate for the changes to SPI config introduced in https://github.com/redhat-appstudio/service-provider-integration-operator/pull/232.

Do not merge this until the SPI PR is also merged and a new SPI release is created.

### CODE REVIEW

File: components/spi/config-patch.json [hunk 1]
```diff
@@ -8,5 +8,15 @@
     "op": "replace",
     "path": "/data/VAULTAUTHMETHOD",
     "value": "kubernetes"
+  },
+  {
+    "op": "replace",
+    "path": "/data/VAULTINSECURETLS",
+    "value": "false"
```
[the remark this answers is not in the document]
Comment (metlos) [reply components/spi/config-patch.json:15]: Yes, that's exactly what it needs :) `VAULTINSECURETLS=false` means we require a trusted TLS cert.

**Comment by Michkov** [comment]
```
kind: ConfigMap
apiVersion: v1
metadata:
  name: spi-shared-config
  namespace: spi-system
  uid: 06175394-cd54-4700-9756-338acacd3ff3
  resourceVersion: '73514'
  creationTimestamp: '2022-08-30T07:13:14Z'
  labels:
    app.kubernetes.io/instance: spi
    app.kubernetes.io/part-of: service-provider-integration-operator
data:
  VAULTAUTHMETHOD: approle
  VAULTHOST: '${VAULT_HOST}'
```

File: components/spi/kustomization.yaml [hunk 2]
```diff
@@ -33,7 +33,7 @@ patches:
     path: oauth-service-deployment-patch.json
   - target:
       kind: ConfigMap
-      name: shared-config
+      name: shared-environment-config
```
Comment (Michkov) [inline components/spi/kustomization.yaml:36]: the configmap name is `spi-shared-config`. When I update it then I got SSL issues:
```
{"level":"error","ts":"2022-08-30T08:19:38Z","caller":"spi-oauth/main.go:126","msg":"failed to create token storage interface","error":"error while authenticating: unable to log in to auth method: unable to log in with Kubernetes auth: Put \"https://spi-vault-spi-system.apps.ci-ln-134278b-72292.origin-ci-int-gce.dev.rhcloud.com/v1/auth/kubernetes/login\": x509: certificate signed by unknown authority","stacktrace":"main.main\n\t/spi-oauth/main.go:126\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250"}
```

**Comment by Michkov** [comment]
KCP version was set in main branch. Closing this PR, in case that this change is critical fix for pre-kcp environment then please re-open the PR against pre-kcp branch. Otherwise integrate your change into kcp version which is in now in main branch.

**Pull request closed without merging**